### PR TITLE
HPCC-19442 Replicate operation on a CSV file causes Segmentation Fault in DFU server.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Fix null pointer deference error.

Add regression suite test case to spray and replicate.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review.

File: testing/regress/ecl/spray_replicate_test.ecl [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import Std.File AS FileServices;
+
+jlib:= SERVICE
+    unsigned8 rtlTick() : library='jlib',eclrtl,entrypoint='rtlNano';
+END;
+
+dropzonePath := '/var/lib/HPCCSystems/mydropzone/' : STORED('dropzonePath');
+espIpPort := 'http://127.0.0.1:8010/FileSpray' : STORED('espIpPort');
+engine := thorlib.platform();
+prefix := engine + '-';
```
Comment (jakesmith) [inline testing/regress/ecl/spray_replicate_test.ecl]: All regression files should be under ~regress:: scope really, so e.g.
```
prefix := '~regress::' + engine + '-'
```

File: testing/regress/ecl/spray_replicate_test.ecl [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+suffix := '-' + jlib.rtlTick() : STORED('timeStamp');
+
+sprayPrepFileName := '~spray_prep';
```
Comment (jakesmith) [inline testing/regress/ecl/spray_replicate_test.ecl]: Similar to above should be under ~regress, could be 
sprayPrepFileName := prefix + 'spray_prep';

File: testing/regress/ecl/spray_replicate_test.ecl [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    desprayOut := output(c1);
+#else
+    desprayOut := output(c1, {result});
+#end
+
+rec spray(rec l) := TRANSFORM
+    SELF.msg := FileServices.fSprayDelimited(
+                        SOURCEIP := '.',
+                        SOURCEPATH := desprayOutFileName,
+                        DESTINATIONGROUP := 'my' + engine,
```
Comment (jakesmith) [inline testing/regress/ecl/spray_replicate_test.ecl]: This won't work in general, e.g. if environment is not completely standard.

Using Std.System.Thorlib.group() should work I think.

**Comment by jakesmith** [comment]
@AttilaVamos - fix looks good, a couple of minor issues with the new test.

**Comment by AttilaVamos** [comment]
@jakesmith updated, please take a look to the test code.

File: testing/regress/ecl/spray_replicate_test.ecl [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+jlib:= SERVICE
+    unsigned8 rtlTick() : library='jlib',eclrtl,entrypoint='rtlNano';
+END;
+
+dropzonePath := '/var/lib/HPCCSystems/mydropzone/' : STORED('dropzonePath');
+espIpPort := 'http://127.0.0.1:8010/FileSpray' : STORED('espIpPort');
+engine := thorlib.platform();
+prefix := '~regress::' + engine + '-';
+suffix := '-' + jlib.rtlTick() : STORED('timeStamp');
```
Comment (jakesmith) [inline testing/regress/ecl/spray_replicate_test.ecl]: was this to make filenames unique?

If so, using + WORKUNIT is the way other tests do that.

**Comment by jakesmith** [comment]
@AttilaVamos - a couple of comments.

**Comment by AttilaVamos** [comment]
@jakesmith changed.

**Review by jakesmith — APPROVED** [review]

**Comment by jakesmith** [comment]
@AttilaVamos - looks good.

@richardkchapman - ready to merge.

**Pull request merged**